File: CS/E4565.Module/BusinessObjects/MyAddress.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Xpo;
using DevExpress.ExpressApp.DC;

namespace E4565.Module.BusinessObjects {
    [DomainComponent]
    public class MyAddress : XPObject {
        public MyAddress(Session session) : base(session) { }
        public string AddressLine {
            get { return GetPropertyValue<string>("AddressLine"); }
            set { SetPropertyValue<string>("AddressLine", value); }
        }
        [Association]
        public MyPerson Person {
            get { return GetPropertyValue<MyPerson>("Person"); }
            set { SetPropertyValue<MyPerson>("Person", value); }
        }
    }
}

File: VB/E4565.Module/BusinessObjects/MyAddress.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp.DC

Namespace E4565.Module.BusinessObjects

    <DomainComponent>
    Public Class MyAddress
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Property AddressLine As String
            Get
                Return GetPropertyValue(Of String)("AddressLine")
            End Get

            Set(ByVal value As String)
                SetPropertyValue(Of String)("AddressLine", value)
            End Set
        End Property

        <Association>
        Public Property Person As MyPerson
            Get
                Return GetPropertyValue(Of MyPerson)("Person")
            End Get

            Set(ByVal value As MyPerson)
                SetPropertyValue(Of MyPerson)("Person", value)
            End Set
        End Property
    End Class
End Namespace
